extensions/SQLServer/SQLServer/SqlServerMemory.cs

```diff
@@ -366,92 +364,104 @@ INNER JOIN
 
     /// <inheritdoc/>
     public async Task<string> UpsertAsync(string index, MemoryRecord record, CancellationToken cancellationToken = default)
+    {
+        return await this.BatchUpsertAsync(index, new[] { record }, cancellationToken);
+    }
+
+    /// <inheritdoc/>
+    public async Task<IEnumerable<string>> BatchUpsertAsync(string index, IEnumerable<MemoryRecord> records, CancellationToken cancellationToken = default)
```

Comment:
the code doesn't build, the method signature doesn't match with the interface

>   SqlServerMemory.cs(19, 43): [CS0738] 'SqlServerMemory' does not implement interface member 'IMemoryDbBatchUpsert.BatchUpsertAsync(string, IEnumerable<MemoryRecord>, CancellationToken)'. 'SqlServerMemory.BatchUpsertAsync(string, IEnumerable<MemoryRecord>, CancellationToken)' cannot implement 'IMemoryDbBatchUpsert.BatchUpsertAsync(string, IEnumerable<MemoryRecord>, CancellationToken)' because it does not have the matching return type of 'IAsyncEnumerable<string>'.